### PR TITLE
Add unit tests for the testing framework

### PR DESCRIPTION
**Pull request by m-sasha**
## Proposed Changes

Add unit tests for the testing framework itself.

## Testing

Test: The PR adds unit tests.

### CODE REVIEW

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            val focusRequester = remember { FocusRequester() }
+            Box(
+                Modifier
+                    .testTag("tag")
+                    .focusRequester(focusRequester)
+                    .focusable()
+            )
+        }
+
+        rule.onNodeWithTag("tag").assertIsNotFocused()
```
Comment (MatkovIvan) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt]: Shouldn't single element be on focus here?

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+            Button(
+                onClick = {},
+                modifier = Modifier.testTag("tag")
+            ) {
+                Box(Modifier.semantics { contentDescription = "desc1" })
+                Box(Modifier.semantics { contentDescription = "desc2" })
+            }
+        }
+
+        rule.onNodeWithTag("tag").assertContentDescriptionEquals("desc1", "desc2")
```
Comment (MatkovIvan) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt]: From `assertContentDescriptionEquals` docs: 
> contains exactly the given values and nothing else.

There is no test that this assert fails if there is something else here.


PS It looks like this is the common problem - there is no tests that it fails when expected

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt (hunk 1 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt]: Nothing requests focus for it, so no.

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt (hunk 2 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/AssertionsTest.kt]: Sure, I'll add failure tests as well.

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**